ADApp/ADSrc/NDArray.cpp

```diff
@@ -119,8 +114,23 @@ int NDArray::getInfo(NDArrayInfo_t *pInfo)
       break;
   }
   pInfo->nElements = 1;
-  for (i=0; i<this->ndims; i++) pInfo->nElements *= this->dims[i].size;
+  for (int i=0; i<ndims; i++) pInfo->nElements *= dims[i];
   pInfo->totalBytes = pInfo->nElements * pInfo->bytesPerElement;
+  return ND_SUCCESS;
+}
+/** Convenience method returns information about an NDArray, including the total number of elements, 
+  * the number of bytes per element, and the total number of bytes in the array.
+  \param[out] pInfo Pointer to an NDArrayInfo_t structure, must have been allocated by caller. */
+int NDArray::getInfo(NDArrayInfo_t *pInfo)
+{
+  int i;
+  NDAttribute *pAttribute;
+  size_t *dims_t = new size_t(this->ndims);
```

Comment:
Instead of allocating memory here with new and then having to manage the deletion could we instead use std::vector on the stack?
Something like

```
std::vector<size_t> dims_t;
for (i=0; i<this->ndims; i++){
  dims_t.push_back(this->dims[i].size);
}
```

Reply:
Review with @ulrikpedersen

Reply:
I've changed to std::vector. It needs to be passed to computeArrayInfo as &dims_t[0], which is a bit ugly.

Reply:
> I've changed to std::vector. It needs to be passed to computeArrayInfo as &dims_t[0], which is a bit ugly.

Ah yes, and if the vector is empty then the behaviour of dims_t[0] is undefined (could lead to some difficult bugs). You could go all-out STL and make computeArrayInfo work with a vector instead of a ptr...

Reply:
I tried using std::array rather than std::vector, which is much cleaner.  But it is only C++11 which we don't want to require (yet!).

Reply:
I think something like this would also work:

``` c++
size_t dims_t[this->ndims];
for (i=0; i<this->ndims; i++)
  dims_t[i] = this->dims[i].size;
```

Reply:
@brunoseivam that (i.e. initializing using a dynamic size: size_t dims_t[variable]) would only work with C++11...

Reply:
It worked with `-std=c++98`, but it might be a gcc-ism...

Reply:
It compiles with gcc 4.8.5 without warning that C++11 is needed.  However, it fails to compile on VS2015 and VS2017.

Reply:
I am reverting to __new__ and __delete__ because I think it is more efficient and legible than the std::vector version.